pkg/controller/master-controller-manager/serviceaccount-projectbinding-controller/controller.go

```diff
@@ -14,89 +14,99 @@ See the License for the specific language governing permissions and
+}
+
+func Add(mgr manager.Manager, log *zap.SugaredLogger) error {
+	r := &reconcileServiceAccountProjectBinding{
+		Client: mgr.GetClient(),
+
+		log: log,
+	}
 	// Create a new controller
 	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
 	if err != nil {
 		return err
 	}
 
 	// Watch for changes to User
-	return c.Watch(&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{})
-}
+	isServiceAccount := predicate.NewPredicateFuncs(func(object ctrlruntimeclient.Object) bool {
+		return kubermaticv1helper.IsProjectServiceAccount(object.GetName())
+	})
 
-// reconcileServiceAccountProjectBinding reconciles User objects.
-type reconcileServiceAccountProjectBinding struct {
-	client.Client
+	if err = c.Watch(&source.Kind{Type: &kubermaticv1.User{}}, &handler.EnqueueRequestForObject{}, isServiceAccount); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 func (r *reconcileServiceAccountProjectBinding) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	resourceName := request.Name
-	// project service account
-	if strings.HasPrefix(resourceName, ServiceAcountName) {
-		if err := r.ensureServiceAccountProjectBinding(ctx, resourceName); err != nil {
-			logger := log.Logger.With("controller", controllerName)
-			logger.Errorw("failed to reconcile in controller", "error", err)
-			return reconcile.Result{}, err
-		}
+	log := r.log.With("serviceaccount", request.Name)
+
+	err := r.ensureServiceAccountProjectBinding(ctx, log, request.Name)
+	if err != nil {
+		log.Errorw("failed to reconcile", zap.Error(err))
 	}
 
-	return reconcile.Result{}, nil
+	return reconcile.Result{}, err
 }
 
-func (r *reconcileServiceAccountProjectBinding) ensureServiceAccountProjectBinding(ctx context.Context, saName string) error {
+func (r *reconcileServiceAccountProjectBinding) ensureServiceAccountProjectBinding(ctx context.Context, log *zap.SugaredLogger, saName string) error {
+	log.Debug("Reconciling")
+
 	sa := &kubermaticv1.User{}
-	if err := r.Get(ctx, client.ObjectKey{Namespace: metav1.NamespaceAll, Name: saName}, sa); err != nil {
+	if err := r.Get(ctx, ctrlruntimeclient.ObjectKey{Name: saName}, sa); err != nil {
 		if kerrors.IsNotFound(err) {
 			return nil
 		}
 		return err
 	}
 
 	// get project name from Owner Reference or from label
-	projectName := ""
-	for _, owner := range sa.GetOwnerReferences() {
-		if owner.APIVersion == kubermaticv1.SchemeGroupVersion.String() && owner.Kind == kubermaticv1.ProjectKindName &&
-			len(owner.Name) > 0 && len(owner.UID) > 0 {
-			projectName = owner.Name
-			break
-		}
-	}
+	projectName := sa.Spec.Project
 	if len(projectName) == 0 {
+		// TODO: Is this fallback still needed?
 		projectName = sa.GetLabels()[kubermaticv1.ProjectIDLabelKey]
 	}
 
 	if len(projectName) == 0 {
-		return fmt.Errorf("unable to find owing project for the service account %q", sa.Name)
+		return errors.New("unable to find owing project")
```

Comment:
while changing this, you can also change `owing->owner`, not sure if owing is the right word for this